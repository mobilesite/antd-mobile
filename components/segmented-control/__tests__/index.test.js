import React from 'react';
import { render, shallow } from 'enzyme';
import { renderToJson } from 'enzyme-to-json';
import SegmentedControl from '../index';

describe('SegmentedControl', () => {
  it('renders correctly', () => {
    const wrapper = render(<SegmentedControl values={['切换一', '切换二']} />);
    expect(renderToJson(wrapper)).toMatchSnapshot();
  });

  it('check api', () => {
    const onValueChange = jest.fn();
    const onChange = jest.fn();
    const wrapper = shallow(
      <SegmentedControl values={['切换一', '切换二']} onValueChange={onValueChange} onChange={onChange} />,
    );
    expect(wrapper.find('.ej-segment-item')).toHaveLength(2);
    expect(wrapper.find('.ej-segment-item').at(0).hasClass('ej-segment-item-selected')).toBeTruthy();
    wrapper.find('.ej-segment-item').at(1).simulate('click', {});
    expect(onValueChange).toHaveBeenCalledWith('切换二');
    expect(onChange).toHaveBeenCalledWith(
      {
        nativeEvent: {
          selectedSegmentIndex: 1,
          value: '切换二',
        },
      },
    );
    expect(wrapper.find('.ej-segment-item').at(0).hasClass('ej-segment-item-selected')).toBeFalsy();
    expect(wrapper.find('.ej-segment-item').at(1).hasClass('ej-segment-item-selected')).toBeTruthy();
  });
});
