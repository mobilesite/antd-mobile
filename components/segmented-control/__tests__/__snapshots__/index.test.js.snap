// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SegmentedControl renders correctly 1`] = `
<div
  class="ej-segment"
  role="tablist"
>
  <div
    aria-disabled="false"
    aria-selected="true"
    class="ej-segment-item ej-segment-item-selected"
    role="tab"
    style="color:#fff;background-color:;border-color:"
  >
    <div
      class="ej-segment-item-inner"
    />
    切换一
  </div>
  <div
    aria-disabled="false"
    aria-selected="false"
    class="ej-segment-item"
    role="tab"
    style="color:;background-color:transparent;border-color:"
  >
    <div
      class="ej-segment-item-inner"
    />
    切换二
  </div>
</div>
`;
